### ISSUE TITLE
Incorrect File Date/Time from rclone serve http

### ISSUE THREAD

**Issue by alan113696** [rclone/rclone#4201]
#### What is the problem you are having with rclone?
When using `rclone serve http` to serve a backend, and then using `rclone copy` to transfer files, the resulting files do not have the proper date/time. The date/time ends up being "Dec 31, 1969" for all files.

#### What is your rclone version (output from `rclone version`)
rclone v1.51.0
- os/arch: linux/amd64
- go version: go1.13.7

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux, 64 bit
CentOS Linux release 7.8.2003 (Core)
3.10.0-1127.el7.x86_64

####  Which cloud storage system are you using? (eg Google Drive)
Tested with S3 and local

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
```
rclone serve http local: --read-only
rclone copy --http-url http://localhost:8080 :http:test ./test -P
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
rclone copy --http-url http://localhost:8080 :http:test ./test -vv --dump requests

2020/05/04 15:52:48 DEBUG : rclone: Version "v1.51.0" starting with parameters ["rclone" "copy" "--http-url" "http://localhost:8080" ":http:test" "./test" "-vv" "--dump" "requests"]
2020/05/04 15:52:48 DEBUG : Using config file from "/home/mydev/.config/rclone/rclone.conf"
2020/05/04 15:52:48 DEBUG : You have specified to dump information. Please be noted that the Accept-Encoding as shown may not be correct in the request and the response may not show Content-Encoding if the go standard libraries auto gzip encoding was in effect. In this case the body of the request will be gunzipped before showing it.
2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : HTTP REQUEST (req 0xc00051a900)
2020/05/04 15:52:48 DEBUG : HEAD /test HTTP/1.1
Host: localhost:8080
User-Agent: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : HTTP RESPONSE (req 0xc00051a900)
2020/05/04 15:52:48 DEBUG : HTTP/1.1 404 Not Found
Content-Length: 11
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Mon, 04 May 2020 19:52:48 GMT
Server: rclone/v1.51.0
X-Content-Type-Options: nosniff

2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : HTTP REQUEST (req 0xc00051af00)
2020/05/04 15:52:48 DEBUG : GET /test/ HTTP/1.1
Host: localhost:8080
User-Agent: rclone/v1.51.0
Accept-Encoding: gzip

2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : HTTP RESPONSE (req 0xc00051af00)
2020/05/04 15:52:48 DEBUG : HTTP/1.1 200 OK
Content-Length: 252
Accept-Ranges: bytes
Content-Type: text/html; charset=utf-8
Date: Mon, 04 May 2020 19:52:48 GMT
Server: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : HTTP REQUEST (req 0xc00051b200)
2020/05/04 15:52:48 DEBUG : HEAD /test/file1.txt HTTP/1.1
Host: localhost:8080
User-Agent: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : HTTP REQUEST (req 0xc00051b400)
2020/05/04 15:52:48 DEBUG : HEAD /test/file2.txt HTTP/1.1
Host: localhost:8080
User-Agent: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : HTTP RESPONSE (req 0xc00051b200)
2020/05/04 15:52:48 DEBUG : HTTP/1.1 200 OK
Content-Length: 6
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Mon, 04 May 2020 19:52:48 GMT
Server: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : HTTP RESPONSE (req 0xc00051b400)
2020/05/04 15:52:48 DEBUG : HTTP/1.1 200 OK
Content-Length: 6
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Mon, 04 May 2020 19:52:48 GMT
Server: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 INFO : Local file system at /tmp/test: Waiting for checks to finish
2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : HTTP REQUEST (req 0xc00051b800)
2020/05/04 15:52:48 DEBUG : GET /test/file1.txt HTTP/1.1
Host: localhost:8080
User-Agent: rclone/v1.51.0
Accept-Encoding: gzip

2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 DEBUG : HTTP REQUEST (req 0xc00051ba00)
2020/05/04 15:52:48 DEBUG : GET /test/file2.txt HTTP/1.1
Host: localhost:8080
User-Agent: rclone/v1.51.0
Accept-Encoding: gzip

2020/05/04 15:52:48 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2020/05/04 15:52:48 INFO : Local file system at /tmp/test: Waiting for transfers to finish
2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : HTTP RESPONSE (req 0xc00051ba00)
2020/05/04 15:52:48 DEBUG : HTTP/1.1 200 OK
Content-Length: 6
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Mon, 04 May 2020 19:52:48 GMT
Last-Modified: Mon, 04 May 2020 19:33:30 GMT
Server: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 DEBUG : HTTP RESPONSE (req 0xc00051b800)
2020/05/04 15:52:48 DEBUG : HTTP/1.1 200 OK
Content-Length: 6
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Mon, 04 May 2020 19:52:48 GMT
Last-Modified: Mon, 04 May 2020 19:33:26 GMT
Server: rclone/v1.51.0

2020/05/04 15:52:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2020/05/04 15:52:48 INFO : file1.txt: Copied (new)
2020/05/04 15:52:48 INFO : file2.txt: Copied (new)
2020/05/04 15:52:48 INFO :
Transferred: 12 / 12 Bytes, 100%, 480 Bytes/s, ETA 0s
Transferred: 2 / 2, 100%
Elapsed time: 0.0s

2020/05/04 15:52:48 DEBUG : 9 go routines active
2020/05/04 15:52:48 DEBUG : rclone: Version "v1.51.0" finishing with parameters ["rclone" "copy" "--http-url" "http://localhost:8080" ":http:test" "./test" "-vv" "--dump" "requests"]
```

**Comment by alan113696** [comment]
[Linking ](https://forum.rclone.org/t/incorrect-file-date-time-from-rclone-serve-http/16108)a discussion about this from the forums.

**Comment by ncw** [comment]
I managed to reproduce that - thanks.

It looks like we forgot to set the Last-Modified header in the http server...

Can you give this a go? It should preserve the timestamps to 1 second accuracy which is what the Last-Modified header allows for.

https://beta.rclone.org/branch/v1.51.0-279-gbdc91eda-fix-4201-httpd-times-beta/ (uploaded in 15-30 mins)

**Comment by calisro** [comment]
Should the same be done for webdav since it serves indexes and could be used the same way.

**Comment by alan113696** [comment]
Works! I tested `rclone serve http` with a local backend as well as S3. Thanks for the quick fix. When will the next version be released?

**Comment by ncw** [comment]
> Should the same be done for webdav since it serves indexes and could be used the same way.

That is done by the webdav library rclone uses and it looks like it is already correct

```
$ curl -I http://127.0.0.1:8080/README.md
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 5822
Content-Type: text/markdown; charset=utf-8
Etag: "160bde3cdb28dabf16be"
Last-Modified: Mon, 04 May 2020 15:54:06 GMT
Date: Tue, 05 May 2020 13:41:47 GMT

$ ls -l README.md 
-rw-rw-r-- 1 ncw ncw 5822 May  4 16:54 README.md
```

**Comment by ncw** [comment]
> Works! I tested `rclone serve http` with a local backend as well as S3. Thanks for the quick fix. When will the next version be released?

Thanks for testing :-)

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52 which should be in the next week or two.
